pkg/networkservice/mechanisms/memif/common.go

```diff
@@ -175,11 +177,15 @@ func create(ctx context.Context, conn *networkservice.Connection, vppConn api.Co
 			}
 			mechanism.SetSocketFileURL((&url.URL{Scheme: memifMech.SocketFileScheme, Path: socketFile(conn)}).String())
 		}
+		mode := memif.MEMIF_MODE_API_ETHERNET
+		if conn.GetPayload() == payload.IP {
+			mode = memif.MEMIF_MODE_API_IP
+		}
```

Comment:
Nicely done... this is simpler than the approach I was taking in my local code :)